sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridSasBuilder.cs

```diff
@@ -0,0 +1,91 @@
+    public class EventGridSasBuilder
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventGridSasBuilder"/> class.
+        /// </summary>
+        /// <param name="endpoint">The endpoint to generate a shared access signature for.
+        /// For example, "https://TOPIC-NAME.REGION-NAME.eventgrid.azure.net/eventGrid/api/events".</param>
+        /// <param name="expiresOn">The time at which the shared access signature should expire.</param>
+        public EventGridSasBuilder(Uri endpoint, DateTimeOffset expiresOn)
+        {
+            Argument.AssertNotNull(endpoint, nameof(endpoint));
+            Endpoint = endpoint;
+            ExpiresOn = expiresOn;
+        }
+
+        /// <summary>
+        /// Gets or sets the endpoint to generate a shared access signature for.
+        /// </summary>
+        public Uri Endpoint
+        {
+            get
+            {
+                return _endpoint;
+            }
+            set
+            {
+                Argument.AssertNotNull(value, nameof(value));
+                _endpoint = value;
+            }
+        }
+
+        private Uri _endpoint;
+
+        /// <summary>
+        /// Gets or sets the time at which the shared access signature should expire.
+        /// </summary>
+        public DateTimeOffset ExpiresOn { get; set; }
+
+        /// <summary>
+        /// Gets or sets the service version to use when generating the shared access signature.
+        /// </summary>
+        public EventGridPublisherClientOptions.ServiceVersion ApiVersion { get; set; } = EventGridPublisherClientOptions.LatestVersion;
+
+        /// <summary>
+        /// Generates a shared access signature that can be used to authenticate with a topic.
+        /// The signature can be used as the input to the <see cref="AzureSasCredential(string)"/> constructor.
+        /// This credential can then be passed to the <see cref="EventGridPublisherClient(Uri, AzureSasCredential, EventGridPublisherClientOptions)"/> constructor.
+        /// </summary>
+        /// <param name="key">The <see cref="AzureKeyCredential"/> to use to authenticate with the service
+        /// when generating the shared access signature.</param>
+        /// <returns>A shared access signature that can be used to authenticate with an Event Grid topic.</returns>
+        public string GenerateSas(AzureKeyCredential key)
+        {
+            Argument.AssertNotNull(key, nameof(key));
+            const char Resource = 'r';
+            const char Expiration = 'e';
+            const char Signature = 's';
+
+            var uriBuilder = new RequestUriBuilder();
+            uriBuilder.Reset(Endpoint);
+            uriBuilder.AppendQuery("api-version", ApiVersion.GetVersionString(), true);
+            string encodedResource = HttpUtility.UrlEncode(uriBuilder.ToString());
+            var encodedExpirationUtc = HttpUtility.UrlEncode(ExpiresOn.ToString(CultureInfo.InvariantCulture));
```

Comment:
This was previously using "en-us" for the culture as that is what was referenced in https://docs.microsoft.com/en-us/azure/event-grid/security-authenticate-publishing-clients#generate-sas-token-programmatically

But InvariantCulture seems to work fine.

Reply:
Slight preference for keeping en-US since that's what was documented by the service, but I agree in practice there is unlikely to be a difference between the two cultures in practice.

Reply:
Fair enough, will revert.